lib/index.js

```diff
@@ -78,36 +90,29 @@ AdRoll.prototype.page = function(page) {
 
 AdRoll.prototype.track = function(track) {
   var event = track.event();
-  var user = this.analytics.user();
   var events = this.events(event);
-  var total = track.revenue() || track.total() || 0;
-  var orderId = track.orderId() || 0;
-  var productId = track.id();
-  var sku = track.sku();
-  var customProps = track.properties();
+  var user = this.analytics.user();
+
+  var data = foldl(function(props, val, key) {
+    props[snake(key)] = val;
+    return props;
+  }, track.properties({
+    revenue: 'adroll_conversion_value_in_dollars',
+    total: 'adroll_conversion_value_in_dollars',
+    orderId: 'order_id',
+    id: 'product_id'
+  }));
 
-  var data = {};
   if (user.id()) data.user_id = user.id();
-  if (orderId) data.order_id = orderId;
-  if (productId) data.product_id = productId;
-  if (sku) data.sku = sku;
-  if (total) data.adroll_conversion_value_in_dollars = total;
-  del(customProps, 'revenue');
-  del(customProps, 'total');
-  del(customProps, 'orderId');
-  del(customProps, 'id');
-  del(customProps, 'sku');
-  if (!is.empty(customProps)) data.adroll_custom_data = customProps;
 
   each(events, function(event) {
-    // the adroll interface only allows for
-    // segment names which are snake cased.
-    data.adroll_segments = snake(event);
+    data.adroll_segments = event;
     window.__adroll.record_user(data);
   });
 
-  // no events found
-  if (!events.length) {
+  // no events found. only send this event if v1, as adroll no longer
+  // accepts segments by name and instead only by segment id
+  if (this.options._version === 1 && !events.length) {
```

Comment:
Let's reformat this into an early return and move it above the `each` call above

```js
if (!events.length) {
  // As of _____ Adroll no longer accepts segments by name, instead tracking by segment ID, which we represent as mapped events or whatever
  // Support legacy customers by sending unmapped events
  if (this.options._version === 1) {
    // make an adroll call
  }
  return;
}
```